File: src/epub_merge_tool/inspect.py
from __future__ import annotations

import zipfile
from pathlib import Path

from .epub_io import MERGE_MANIFEST_PATH, require_manifest
from .errors import ManifestError


def inspect_epub(path: Path | str) -> dict:
    with zipfile.ZipFile(Path(path).expanduser(), "r") as zf:
        try:
            manifest = require_manifest(zf.read(MERGE_MANIFEST_PATH))
        except KeyError as exc:
            raise ManifestError("missing epub-merge-tool manifest") from exc
    return {
        "tool_generated": True,
        "schema": manifest["schema"],
        "structure": manifest["structure"],
        "title": manifest["title"],
        "sources": [source["basename"] for source in manifest["sources"]],
    }

File: ts/src/inspect.ts
import { ManifestError } from "./errors.js";
import { MERGE_MANIFEST_PATH, requireMergeManifest } from "./epub-io.js";
import { readZip } from "./zip.js";

export interface InspectReport {
  tool_generated: true;
  schema: string;
  structure: string;
  title: string;
  sources: string[];
}

export function inspectEpub(input: Uint8Array): InspectReport {
  const zip = readZip(input);
  const manifestData = zip.get(MERGE_MANIFEST_PATH);
  if (!manifestData) throw new ManifestError("missing epub-merge-tool manifest");
  const manifest = requireMergeManifest(manifestData);
  const sources = manifest.sources as Array<Record<string, unknown>>;
  return {
    tool_generated: true,
    schema: String(manifest.schema),
    structure: String(manifest.structure),
    title: String(manifest.title),
    sources: sources.map((source) => String(source.basename))
  };
}
